src/services/formatting/smartIndenter.ts

```diff
@@ -464,9 +471,29 @@ namespace ts.formatting {
                 case SyntaxKind.GetAccessor:
                 case SyntaxKind.SetAccessor:
                     return child !== SyntaxKind.Block;
+                case SyntaxKind.ExportDeclaration:
+                    return child !== SyntaxKind.NamedExports;
                 default:
                     return false;
             }
         }
+
+        function hasBraceShell(node: Node) {

```

Comment:
Not sure why this is needed? Can we just do what is done for all other kinds of blocks, namely:
- add `SyntaxKind.NamedImports`, `SyntaxKind.NamedExports` to the list in `shouldIndentChildNode` function so its content will be indented
- update `isCompletedNode` function to correctly process case when cursor is in incompleted `NamesImport` / `NamedExport` node by using `hasChildOfKind(n, SyntaxKind.CloseBraceToken, sourceFile);`